src/firefly/java/edu/caltech/ipac/util/StringUtils.java

```diff
@@ -63,6 +63,15 @@
         }
     }
 
+    /**
+     * applies the consumer's logic if v is not null or is an empty string
+     * @param v a value to check against
+     * @param f the consumer function to apply if v is not empty
+     */
+    public static <T> void applyIfNotEmpty(T v, Consumer<T> f){
```

Comment:
This is very useful function! I guess it takes us to move  to Java10 to start using Java 8 features.

Reply:
I'm curious, what is exactly the newest feature here compare to jdk7?

Reply:
generics is old, since jdk 1.5.  but with lambda in jdk 1.8, it allows you to write a single-method interface like this:

`  applyIfNotEmpty(something.getName(), v -> aMap.put("name", v));`